python/hail/methods/impex.py

```diff
@@ -764,13 +766,183 @@ def import_gen(path, sample_file=None, tolerance=0.2, min_partitions=None, chrom
+
+    In this case, we should:
+
+    - Pass the non-default delimiter ``,``
+
+    - Pass the non-default missing value ``.``
+
+    >>> table = methods.import_table('data/samples2.tsv', delimiter=',', missing='.')
+
+    Let's import a table from a file with no header and sample IDs that need to
+    be transformed.  Suppose the sample IDs are of the form ``NA#####``. This
+    file has no header line, and the sample ID is hidden in a field with other
+    information.
+
+    .. code-block: text
+
+        $ cat data/samples3.tsv
+        1kg_NA12345   female
+        1kg_NA12346   male
+        1kg_NA12348   female
+        pgc_NA23415   male
+        pgc_NA23418   male
+
+    To import:
+
+    >>> t = (methods.import_table('data/samples3.tsv', no_header=True)
+    ...      .annotate('sample = f0.split("_")[1]')
+    ...      .key_by('sample'))
+
+    Notes
+    -----
+
+    The `impute` parameter tells Hail to scan the file an extra time to gather
+    information about possible field types. While this is a bit slower for large
+    files because the file is parsed twice, the convenience is often worth this
+    cost.
+
+    The `delimiter` parameter is either a delimiter character (if a single
+    character) or a field separator regex (2 or more characters). This regex
+    follows the `Java regex standard
+    <http://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html>`_.
+
+    .. note::
+
+        Use ``delimiter='\\s+'`` to specify whitespace delimited files.
+
+    If set, the `comment` parameter causes Hail to skip any line that starts
+    with the given string. For example, passing ``comment='#'`` will skip any
+    line beginning in a pound sign.
+
+    The `missing` parameter defines the representation of missing data in the table.
+
+    .. note::
+
+        The `comment` and `missing` parameters are **NOT** regexes.
+
+    The `no_header` parameter indicates that the file has no header line. If
+    this option is passed, then the field names will be `f0`, `f1`,
+    ... `fN` (0-indexed).
+
+    The `types` parameter allows the user to pass the types of files in the
+    table. It is an :obj:`dict` keyed by :obj:`str`, with :class:`.Type`
+    values. See the examples above for a standard usage. Additionally, this
+    option can be used to override type imputation. For example, if the field
+    ``Chromosome`` only contains the values ``1`` through ``22``, it will be
+    imputed to have type :class:.TInt32, whereas most Hail methods expect that a
+    chromosome field will be of type :class:.TString.  Setting ``impute=True``
+    and ``types={'Chromosome': TString()}`` solves this problem.
+
+    The `min_partitions` parameter sets the minimum number of partitions (level
+    of sharding) of an imported table. The default partition size depends on
+    file system and a number of other factors (including the `min_block_size`
+    parameter of :meth:`.HailContext.init`), but usually is between 32M and
```

Comment:
hail context

Reply:
? How do I set the `min_block_size` without calling `HailContext.init`?

Reply:
hail context is gone! just link to ~hail.init

Reply:
sorry, should have been more descriptive